app/routes.py

```diff
@@ -1,2 +1,254 @@
+from app.models.customer import Customer
+from app.models.video import Video
+from app.models.rental import Rental
+from app import db
+import requests
+from datetime import datetime, timedelta
+from dotenv import load_dotenv
 
+load_dotenv()
+
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+@customers_bp.route("", methods=["POST"], strict_slashes=False)
+def post_customer():
+    request_body = request.get_json()
+    if "name" not in request_body or "postal_code" not in request_body or "phone" not in request_body:
+        return make_response({
+            "details": "Invalid data"
+        }, 400)
+    else:
+        new_customer = Customer(name=request_body["name"],
+                                postal_code=request_body["postal_code"],
+                                phone=request_body["phone"])
+        db.session.add(new_customer)
+        db.session.commit()
+        
+        response = new_customer.return_data()
+
+        return(make_response(response), 201)
+
+@customers_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_customers():
+    customers = Customer.query.all()
+
+    customers_response = []
+    for customer in customers:
+        customers_response.append(customer.return_data())
+    
+    return (make_response(jsonify(customers_response)), 200)
+
+@customers_bp.route("/<customer_id>", methods=["GET"], strict_slashes=False)
+def get_one_customer(customer_id):
+    customer = Customer.query.get_or_404(customer_id)
+
+    response = customer.return_data()
+    return make_response((response), 200)
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"], strict_slashes=False)
+def get_customers_rentals(customer_id):
+    customer = Customer.query.get_or_404(customer_id)
+    response= []
+    for rental in customer.rentals_out:
+        video = Video.query.get_or_404(rental.video_id)
+        response.append(
+            {"release_date": video.release_date,
+            "title": video.title,
+            "due_date":rental.due_date
+        })
+    return make_response(jsonify(response), 200)
+
+@customers_bp.route("/<customer_id>", methods=["PUT"], strict_slashes=False)
+def update_customer(customer_id):
+    customer = Customer.query.get_or_404(customer_id)
+    request_body = request.get_json()
+    print(f"request body: {request_body}")
+
+    if "name" not in request_body \
+        or "postal_code" not in request_body \
+        or "phone" not in request_body:
+        return make_response({}, 400)
+        
+    elif type(request_body["name"]) is not str or request_body["name"] == "" \
+        or type(request_body["phone"]) is not str or request_body["phone"] == "" \
+        or type(request_body["postal_code"]) is not int or request_body["postal_code"] == "":
+        return make_response({}, 400)
```

Comment:
since these have the same return statement, maybe we can combine the two conditionals and they can share the same return statement